gensim/models/sent2vec.py

```diff
@@ -0,0 +1,793 @@
+        the model's cached `epochs` value should be supplied as `epochs` value.
+
+        Parameters
+        ----------
+        sentences : iterable of iterables
+            The `sentences` iterable can be simply a list of lists of tokens, but for larger corpora,
+            consider an iterable that streams the sentences directly from disk/network.
+        total_examples : int
+            Count of sentences.
+        total_words : int
+            Count of raw words in sentences.
+        epochs : int
+            Number of iterations (epochs) over the corpus.
+        start_alpha : float
+            Initial learning rate.
+        end_alpha : float
+            Final learning rate. Drops linearly from `start_alpha`.
+        word_count : int
+            Count of words already trained. Set this to 0 for the usual
+            case of training on all words in sentences.
+        queue_factor : int
+            Multiplier for size of queue (number of workers * queue_factor).
+        report_delay : float
+            Seconds to wait before reporting progress.
+        compute_loss: bool
+            If True, computes and stores loss value which can be retrieved using `model.get_latest_training_loss()`.
+        callbacks : :obj: `list` of :obj: `~gensim.models.callbacks.CallbackAny2Vec`
+            List of callbacks that need to be executed/run at specific stages during training.
+
+        Examples
+        --------
+        >>> from gensim.models import Sent2Vec
+        >>> sentences = [["cat", "say", "meow"], ["dog", "say", "woof"]]
+        >>>
+        >>> model = Sent2Vec(min_count=1)
+        >>> model.build_vocab(sentences)
+        >>> model.train(sentences, total_examples=model.corpus_count, epochs=model.epochs)
+        """
+
+        if FAST_VERSION < 0:
+            return self._do_train_job_slow(sentences)
+        return super(Sent2Vec, self).train(
+            sentences, total_examples=total_examples, total_words=total_words,
+            epochs=epochs, start_alpha=start_alpha, end_alpha=end_alpha, word_count=word_count,
+            queue_factor=queue_factor, report_delay=report_delay, compute_loss=compute_loss, callbacks=callbacks)
+
+    def __getitem__(self, sentence):
+        """Get sentence vector for an input sentence.
+
+        Parameters
+        ----------
+        sentence : list of str
+            List of words.
+
+        Returns
+        -------
+        numpy.ndarray
+            Sentence vector for input sentence.
+
+        """
+
+        ntokens_temp, words = self.vocabulary.get_line(sentence)
+        sent_vec = np.zeros(self.vector_size)
+        line = self.vocabulary.add_ngrams(context=words, n=self.word_ngrams)
+        for word_vec in line:
+            sent_vec += self.wi[word_vec]
+        if len(line) > 0:
+            sent_vec *= (1.0 / len(line))
+        return sent_vec
+
+    @classmethod
+    def load(cls, *args, **kwargs):
```

Comment:
remove this method

Reply:
Can't remove this. I need to call `utils.SaveLoad` load function instead of `BaseWordEmbeddingsModel`.